src/java.base/share/classes/java/text/NumberFormat.java

```diff
@@ -59,102 +59,108 @@
+ * For example, {@code "2K"}.
+ * <li> Use {@link #getPercentInstance} to get a format for displaying percentages.
+ * For example, {@code "200,050%"}.
+ * </ul>
  *
- * <p>
- * To format a number for the current Locale, use one of the factory
- * class methods:
- * <blockquote>
- * {@snippet lang=java :
- * myString = NumberFormat.getInstance().format(myNumber);
- * }
- * </blockquote>
- * If you are formatting multiple numbers, it is
- * more efficient to get the format and use it multiple times so that
- * the system doesn't have to fetch the information about the local
- * language and country conventions multiple times.
- * <blockquote>
- * {@snippet lang=java :
- * NumberFormat nf = NumberFormat.getInstance();
- * for (var myNumber : numbers) {
- *     output.println(nf.format(myNumber) + "; ");
- * }
- * }
- * </blockquote>
- * To format a number for a different Locale, specify it in the
- * call to {@code getInstance}.
- * <blockquote>
- * {@snippet lang=java :
- * NumberFormat nf = NumberFormat.getInstance(Locale.FRENCH);
- * }
- * </blockquote>
+ * Alternatively, if a {@code NumberFormat} for a different locale is required, use
+ * one of the overloaded factory methods that take {@code Locale} as a parameter,
+ * for example, {@link #getIntegerInstance(Locale)}. If the installed locale-sensitive
+ * service implementation does not support the given {@code Locale}, {@link Locale#ROOT}
+ * will be used as the fallback {@code Locale}.
  *
- * <p>If the locale contains "nu" (numbers) and/or "rg" (region override)
+ * <h3>Locale Extensions</h3>
+ * Formatting behavior can be changed when using a locale that contains any of the following
  * <a href="../util/Locale.html#def_locale_extension">Unicode extensions</a>,
- * the decimal digits, and/or the country used for formatting are overridden.
+ * <ul>
+ * <li> "nu"
+ * (<a href="https://unicode.org/reports/tr35/#UnicodeNumberSystemIdentifier">
+ * Numbering System</a>) - Overrides the decimal digits used
+ * <li> "rg"
+ * (<a href="https://unicode.org/reports/tr35/#RegionOverride">
+ * Region Override</a>) - Overrides the country used
+ * <li> "cf"
+ * (<a href="https://www.unicode.org/reports/tr35/tr35.html#UnicodeCurrencyFormatIdentifier">
+ * Currency Format style</a>) - Overrides the Currency Format style used
+ * </ul>
+ * <p>
  * If both "nu" and "rg" are specified, the decimal digits from the "nu"
  * extension supersedes the implicit one from the "rg" extension.
+ * Although <a href="../util/Locale.html#def_locale_extension">Unicode extensions</a>
+ * defines various keys and values, actual locale-sensitive service implementations
+ * in a Java Runtime Environment might not support any particular Unicode locale
+ * attributes or key/type pairs.
+ * <p>Below is an example of a "US" locale currency format with accounting style,
+ * <blockquote>{@code NumberFormat.getInstance(Locale.forLanguageTag("en-US-u-cf-account"));}</blockquote>
```

Comment:
I replaced the previous Thai digits example with an account currency format style, as it is probably a more common use case.